integration/smoke/run_smoke_parallel.sh

```diff
@@ -0,0 +1,166 @@
+    #
+    # TODO: This function has some copied code from
+    # integration/experiment/gen_slurm.sh
+    #
+    # Inputs:
+    #   NUM_NODES: number of nodes to use for the job
+    #   APP: name of the application in integration/apps
+    #   EXP_TYPE: experiment type string
+    #
+    # Returns:
+    #   SBATCH_NAME: name of the sbatch script to be submitted, or
+    #                empty if the run script is missing
+    #   RUN_SCRIPT: name of the experiment run script for the given
+    #               app and experiment type
+
+    local EXP_DIR=${EXP_TYPE}
+    if [ "${EXP_TYPE}" == "barrier_frequency_sweep" ] ||
+       [ "${EXP_TYPE}" == "power_balancer_energy" ]; then
+        EXP_DIR="energy_efficiency"
+    fi
+
+    local EXP_ARGS=""
+    if [ "${EXP_TYPE}" == "power_sweep" ] ||
+       [ "${EXP_TYPE}" == "power_balancer_energy" ]; then
+        EXP_ARGS="--min-power=190 --max-power=230"
+    elif [ "${EXP_TYPE}" == "frequency_sweep" ] ||
+         [ "${EXP_TYPE}" == "barrier_frequency_sweep" ]; then
+        EXP_ARGS="--min-frequency=1.9e9 --max-frequency=2.0e9"
+    elif [ "${EXP_TYPE}" == "uncore_frequency_sweep" ]; then
+        EXP_ARGS="--min-frequency=1.9e9 --max-frequency=2.0e9 --min-uncore-frequency=2.1e9 --max-uncore-frequency=2.2e9"
+    fi
+
+    # set up commands for PASS_RESULT, FAIL_RESULT, and SKIP_RESULT
+    save_test_result_line
+
+    RUN_SCRIPT=${GEOPM_SOURCE}/integration/experiment/${EXP_DIR}/run_${EXP_TYPE}_${APP}.py
+    if [ -f "${RUN_SCRIPT}" ]; then
+        SBATCH_NAME=${APP}_${EXP_TYPE}_${NUM_NODES}.sbatch
+        cat > ${SBATCH_NAME} << EOF
+#!/bin/bash
+#SBATCH -N ${NUM_NODES}
+#SBATCH -o %j.out
+#SBATCH -J ${APP}_${EXP_TYPE}
+#SBATCH -t 00:30:00
+
+source ${GEOPM_SOURCE}/integration/config/run_env.sh
+OUTPUT_DIR=\${SLURM_JOBID}_\${SLURM_JOB_NAME}
+
+${RUN_SCRIPT} \\
+    --node-count=\${SLURM_NNODES} \\
+    --output-dir=\${OUTPUT_DIR} \\
+    ${EXP_ARGS} \\
+    # end
+
+result=\$?
+if [ \$result -eq 0 ]; then
+   ${PASS_RESULT}
+else
+   ${FAIL_RESULT}
+fi
+
+EOF
+        # Remove blank lines
+        uniq ${SBATCH_NAME} .${SBATCH_NAME}
+        mv .${SBATCH_NAME} ${SBATCH_NAME}
+    else
+        SBATCH_NAME=""
+    fi
+}
+
+for EXP_TYPE in $ALL_EXP; do
+
+    NUM_NODES=2
+    for APP in ${APPS_2NODE}; do
+        gen_sbatch
+        if [ ! -z $SBATCH_NAME ]; then
+            sbatch ${SBATCH_NAME}
+            echo "${SBATCH_NAME} submitted"
+        else
+            echo "${RUN_SCRIPT} not found; skipping"
+            # TODO: danger!
+            eval ${SKIP_RESULT}
+        fi
+    done
+
+    NUM_NODES=1
+    for APP in ${APPS_1NODE}; do
+        gen_sbatch
+        if [ ! -z $SBATCH_NAME ]; then
+            sbatch ${SBATCH_NAME}
+            echo "${SBATCH_NAME} submitted"
+        else
+            echo "${RUN_SCRIPT} not found; skipping"
+            # TODO: danger!
+            eval ${SKIP_RESULT}
+        fi
+    done
```

Comment:
Seems like this block is repeated from above.  It's probably a good candidate for an extract refactor into a function.